### PR TITLE
Resolve recently introduced warnings

### PR DESCRIPTION
**Pull request by tshortli**
Resolve various warnings that were introduced recently into the compiler and stdlib builds. See individual commits for specific justifications.

### CODE REVIEW

**Comment by tshortli** [comment]
@swift-ci please smoke test

**Comment by tshortli** [comment]
@swift-ci please build toolchain Linux platform

**Comment by tshortli** [comment]
@swift-ci please build toolchain Windows platform

**Comment by tshortli** [comment]
@swift-ci please build toolchain macOS platform

File: lib/AST/InlinableText.cpp [hunk 1]
```diff
@@ -192,6 +192,14 @@ struct ExtractInactiveRanges : public ASTWalker {
 };
 } // end anonymous namespace
 
+#if SWIFT_BUILD_SWIFT_SYNTAX
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wreturn-type-c-linkage"
+extern "C" BridgedStringRef
+swift_ASTGen_extractInlinableText(BridgedASTContext ctx,
```
Comment (DougGregor) [inline lib/AST/InlinableText.cpp:199]: Thank you!

**Review by DougGregor — APPROVED** [review]
Looks great, thank you!

**Review by xymus — APPROVED** [review]

**Pull request merged**